kairon/events/utility.py

```
@@ -32,3 +37,35 @@ def update_job(event_type: Text, request_data: Dict, is_scheduled: bool):
+        except Exception as e:
+            raise AppException(f"Failed to schedule mail reading for bot {bot}. Error: {str(e)}")
+
+    @staticmethod
+    def reschedule_all_bots_channel_mail_reading():
+        try:
+            bots = list(Channels.objects(connector_type= ChannelTypes.MAIL.value).distinct("bot"))
+            for bot in bots:
+                logger.info(f"Rescheduling mail reading for bot {bot}")
+                EventUtility.schedule_channel_mail_reading(bot)
+        except Exception as e:
+            raise AppException(f"Failed to reschedule mail reading events. Error: {str(e)}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)